### PR TITLE
MAML example

### PR DESCRIPTION
**Pull request by fabianp**

### CODE REVIEW

**Comment by fabianp** [comment]
TODO: 

- [ ] Frame it as a bi-level problem, as the implicit maml paper (https://arxiv.org/pdf/1909.04630.pdf)

**Comment by fabianp** [comment]
I think this is finally ready for review 🎉 !

**Comment by fabianp** [comment]
one can see the notebook here: https://github.com/google/jaxopt/blob/d7fc1d5407c81ab3c70bb09b818101fbcb4e1e3e/docs/notebooks/implicit_diff/maml.ipynb

**Comment by fabianp** [comment]
There are still a couple of issues left, but I believe these can be tackled after merge:

1. The notebook takes a while to run, like a couple of hours (on CPU). Would be nice to bring that down.
2. Convergence of the outer gradient seems to plateau around 0.05, not sure why

**Comment by fabianp** [comment]
@asteroidhouse might also be interested by this 😄

File: docs/notebooks/implicit_diff/maml.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+import jax
+from jax import numpy as jnp
+from jax import random
+
+from jaxopt import LBFGS
+from jaxopt import OptaxSolver
+from jaxopt import tree_util
+
+# we use haiku to construct a small multi-layer perceptron
+import haiku as hk
```
Comment (mblondel) [inline docs/notebooks/implicit_diff/maml.md:37]: not needed

File: docs/notebooks/implicit_diff/maml.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+prediction_without_regularization = model.apply(params_without_regularization, xx.reshape((-1, 1)))
+
+plt.plot(xx, prediction.ravel(), lw=3, label="with meta-learning")
+plt.plot(xx, prediction_without_regularization.ravel(), lw=3, label="without meta-learning")
+plt.scatter(train[0], train[1], marker='^', s=80, label=f"Training samples", color=colors[task])
+plt.plot(xs, ys, lw=3, label="ground truth")
+plt.ylim((-1, 1))
+plt.xlim(MIN_X, MAX_X)
+plt.legend(loc='upper center', bbox_to_anchor=(0.5, -0.1), frameon=False, ncol=n_tasks)
+plt.grid()
```
Comment (mblondel) [inline docs/notebooks/implicit_diff/maml.md:269]: To convey the difficulty of the task I would add something like `plt.title("Fit with only 3 samples")`

**Review by mblondel — APPROVED** [review]
LGTM apart from my 2 minor comments

File: docs/notebooks/implicit_diff/maml.md (hunk 2 above)
Comment (fabianp) [reply docs/notebooks/implicit_diff/maml.md:269]: nice suggestion! implemented

**Comment by fabianp** [comment]
thanks for the review! Implemented the suggestions. Please add the pull-ready label if you're satisfied :-)

**Pull request merged**